buildSrc/src/main/groovy/Docker.groovy

```diff
@@ -155,6 +155,10 @@ class Docker {
          */
         boolean showLogsOnSuccess;
     }
+    static class TaskDependencies {
```

Comment:
This needs docs - but the fields are both Object so that they can be a task, taskprovider, or list of the previous two, letting consumers pick they model they want. Might be cleaner to make a full DSL for these, but that gets wordy, so keeping it simple for the moment.